tapdb/post_migration_checks.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	chainAssets, err := dbAssetsToChainAssets(
+		dbAssets, assetWitnesses, defaultClock,
+	)
+	if err != nil {
+		return fmt.Errorf("error converting assets: %w", err)
+	}
+
+	burnAssets := fn.Filter(chainAssets, func(a *asset.ChainAsset) bool {
+		return a.IsBurn()
+	})
+	burnKeys := make(map[asset.SerializedKey]struct{})
+	for _, a := range burnAssets {
+		serializedKey := asset.ToSerialized(a.ScriptKey.PubKey)
+		burnKeys[serializedKey] = struct{}{}
+	}
+
+	untypedKeys, err := q.FetchUnknownTypeScriptKeys(ctx)
+	if err != nil {
+		return fmt.Errorf("error fetching script keys: %w", err)
+	}
+
+	channelFundingKey := asset.NewScriptKey(
+		tapscript.NewChannelFundingScriptTree().TaprootKey,
+	).PubKey
+
+	for _, k := range untypedKeys {
+		scriptKey, err := parseScriptKey(k.InternalKey, k.ScriptKey)
+		if err != nil {
+			return fmt.Errorf("error parsing script key: %w", err)
+		}
+
+		// We ignore the parity bit when comparing it to other keys, as
+		// we generally don't know how it might be stored in the DB.
+		schnorrScriptKey, _ := schnorr.ParsePubKey(
+			schnorr.SerializePubKey(scriptKey.PubKey),
+		)
+
+		serializedKey := asset.ToSerialized(scriptKey.PubKey)
+		newType := asset.ScriptKeyUnknown
+
+		if _, ok := burnKeys[serializedKey]; ok {
+			newType = asset.ScriptKeyBurn
+		} else {
+			assumedType := scriptKey.DetermineType()
+
+			switch {
+			// If we're sure that a key is BIP-86 or the well-known
+			// NUMS (tombstone) key, we mark it as such.
+			case assumedType == asset.ScriptKeyBip86 ||
+				assumedType == asset.ScriptKeyTombstone:
+
+				newType = assumedType
+
+			// Previous channel funding script keys (OP_TRUE) can
+			// be detected, since they are the same key. New, unique
+			// script keys for grouped asset channels are only
+			// introduced with the same version as this migration
+			// ships in, so they should be stored with the correct
+			// type already.
+			case assumedType == asset.ScriptKeyScriptPathExternal &&
+				schnorrScriptKey.IsEqual(channelFundingKey):
+
+				newType = asset.ScriptKeyScriptPathChannel
+
+			// We'll want to not show scripted keys by default in
+			// the balances. So any key that is not a burn key and
+			// not a channel funding key, but has a script path, we
+			// mark as external script path.
+			case assumedType == asset.ScriptKeyScriptPathExternal:
+				newType = asset.ScriptKeyScriptPathExternal
+			}
+		}
+
+		// If we were able to identify the key type, we update the key
+		// in the database.
+		if newType != asset.ScriptKeyUnknown {
+			_, err := q.UpsertScriptKey(ctx, NewScriptKey{
+				InternalKeyID:    k.InternalKey.KeyID,
+				TweakedScriptKey: k.ScriptKey.TweakedScriptKey,
+				Tweak:            k.ScriptKey.Tweak,
+				DeclaredKnown:    k.ScriptKey.DeclaredKnown,
+				KeyType:          sqlInt16(newType),
+			})
+			if err != nil {
+				return fmt.Errorf("error updating script key "+
+					"type: %w", err)
+			}
+		}
```

Comment:
Log if we aren't able to detect the key type? 

Don't think we have a dry run migration mode right now, but may be something we want to add, as can be an extra sanity check here.